### ISSUE TITLE
Clarify wording for floating-point params passing

### ISSUE THREAD

**Issue by FOODzee** [riscv/riscv-elf-psabi-doc#145]
Currently it is not very clear whether the following passage
>A real floating-point argument is passed in a floating-point argument register if it is no more than FLEN bits wide and at least one floating-point argument register is available. Otherwise, it is passed according to the integer calling convention.

from https://github.com/riscv/riscv-elf-psabi-doc/blob/master/riscv-elf.md#hardware-floating-point-calling-convention prescribes to use available `a`-registers instead of stack in case when all `fa`-registers are occupied by floating-point params and there is still more of those.

I've checked the following example using `gcc` and it really passes `foo` parameters as their names predict.
```c
void foo(int a0, float fa0, int a1, float fa1, float fa2, int a2, float fa3, float fa4, float fa5, float fa6, float fa7, float a3, int a4, float a5, int a6, int a7, float stack0, int stack1) {} 

int main() {
foo(0, 1.0, 1, 1.1, 1.2, 2, 1.3, 1.4, 1.5, 1.6, 1.7, 2.8, 4, 2.9, 6, 7, 3.0, 10);
return 0;
}
```

I guess it is somewhat related to #87 .

**Comment by aswaterman** [comment]
I don't think there's any technical ambiguity: your example conforms to my parsimonious reading of the spec.

We can add examples if you want, but I think adding new normative text would only confuse the situation.

**Comment by FOODzee** [comment]
Yeah, I think a couple of examples or pseudocode of regs/stack assignment (like in AArch64) would be great.
